pkg/controller/csr_check.go

```diff
@@ -373,9 +373,9 @@ func authorizeServingCertWithMachine(machines []machinehandlerpkg.Machine, req *
 		var attemptedAddresses []string
 		var foundSan bool
 		for _, addr := range targetMachine.Status.Addresses {
-			switch corev1.NodeAddressType(addr.Type) {
+			switch addr.Type {
```

Comment:
This was an unrelated clean up I found while stepping through the code - unnecessary conversion that can be undone.  Let me know if there's any reason we should leave this as it was.